Update docs for grain-lang/grain#208 (#77)
grain-lang/grain-lang.org: src/stdlib/list.md

```

Evaluates the given function for each item in the list. The given function is called with the element and its index in the list.

### List.**filter**

```grain
filter: (a -> Bool, List<a>) -> List<a>
```

> `List.filter(fn, list)` produces a new list by calling `fn` on each element of the list. If the `fn` returns `false`, the value will be removed from the new list.

Reviewer: ```suggestion
`List.filter(fn, list)` produces a new list by calling `fn` on each element of the list. If the `fn` returns `false`, the value will not be included the new list.
```

### List.**reject**

```grain
reject: (a -> Bool, List<a>) -> List<a>
```

> `List.reject(fn, list)` produces a new list by calling `fn` on each element of the list. If the `fn` returns `true`, the value will be removed from the new list.

Reviewer: ```suggestion
`List.reject(fn, list)` produces a new list by calling `fn` on each element of the list. If the `fn` returns `true`, the value will not be included in the new list.
```

### List.**head**

```grain
head : List<a> -> a
```

Attempts to get the first element from the list. Fails if the list is empty.

### List.**tail**

```grain
tail : List<a> -> List<a>
```

> Attempts to get all elements in a list except the first element. Fails if the list is empty.

Reviewer: ```suggestion
Attempts to get all elements in a list except the first element. Fails if the list has one or fewer elements.
```

Reviewer: My brain stopped working when I wrote this—disregard.

### List.**nth**

```grain
nth : (Number, List<a>) -> List<a>
```

Attempts to get the element in the list at the index provided. Fails if the index is out-of-bounds or if the list is empty.

### List.**flatten**

```grain
flatten : List<List<a>> -> List<a>
```

Turns a list of lists of values into a "flattened" list of the values.

### List.**insert**

```grain
insert : (a, Number, List<a>) -> List<a>
```

Attempts to insert a value into the input list at the given index. Fails if the index is out-of-bounds.

### List.**count**

```grain
count : (a -> Bool, List<a>) -> Number
```

> Counts the number of elements in a list satisfy the condition.

Reviewer: ```suggestion
Counts the number of elements in a list that satisfy the given condition.
```

### List.**part**

```grain
part : (Number, List<a>) -> (List<a>, List<a>)
```

> Splits a list into two lists: the head of `count` elements, and the remaining elements. Fails if the count is out-of-bounds.

Reviewer: ```suggestion
Splits a list into two lists: a list containing the first `count` elements, and a list containing the remaining elements. Fails if the count is out-of-bounds.
```

### List.**rotate**

```grain
rotate : (Number, List<a>) -> List<a>
```

> Produces a new list where the `count` elements are moved to the end of the list.

Reviewer: ```suggestion
Produces a new list where the `count` elements are moved to the end of the list. If a negative count is provided, moves the last `-count` elements to the beginning of the list.
```

### List.**unique**

```grain
unique : List<a> -> List<a>
```

> Produces a new list with any duplicates removed. Uses the generic `==` equality operator.

Reviewer: ```suggestion
Produces a new list with any duplicates removed. Uses the generic `==` structural equality operator.
```
